### DISCUSSION TITLE
nameToFavoriteNumber Explanation

### DISCUSSION THREAD

**Discussion by maxdporter** [smartcontractkit/full-blockchain-solidity-course-py#705] [Q&A] [answered]
In the very last line of the SimpleStorage.sol file, there's this line of code:

"nameToFavoriteNumber[_name] = _favoriteNumber;"

can someone explain how what this line means and how it works?

**Comment by cromewar** [comment, marked as answer]
Hello @maxdporter 
nameToFavoriteNumber is a mapping which links to type of data (string and unit) so you are assigning it using the `_name` as the index and the  `_favoriteNumber` as input. Look at this: https://medium.com/upstate-interactive/mappings-in-solidity-explained-in-under-two-minutes-ecba88aff96e i would help you a lot to understand how it works.

**Reply by maxdporter** [reply]
Thank you! That's pretty straightforward and your article was very helpful.
